src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -223,6 +230,7 @@ public SlaveTemplate(String ami, String zone, SpotConfiguration spotConfig, Stri
         this.useDedicatedTenancy = useDedicatedTenancy;
         this.connectBySSHProcess = connectBySSHProcess;
         this.maxTotalUses = maxTotalUses;
+        this.nodeProperties = new DescribableList<NodeProperty<?>,NodePropertyDescriptor>(Saveable.NOOP, nodeProperties);
```

Comment:
- the type parameters on the constructor are redundant

- the `nodeProperties` argument may be `null` (e.g. in unit tests), this causes a NullPointerException in the constructor of the DescribableList. Wrapping the argument with `Util.fixNull(...)` should prevent this.

Reply:
Done